proxy/proxy.go

```diff
@@ -320,6 +324,10 @@ func WithParams(p Params) *Proxy {
 		p.MaxLoopbacks = 0
 	}
 
+	if p.DefaultHTTPStatus >= http.StatusContinue && p.DefaultHTTPStatus <= http.StatusNetworkAuthenticationRequired {
+		errRouteLookupFailed.code = p.DefaultHTTPStatus
```

Comment:
It feels a bit wrong to modify the package level `errRouteLookupFailed` variable as a side-effect here. Alternatively I could add a `DefaultHTTPStatus` field on the proxy struct and reference that, when the error occur.

What do you think?

Reply:
agree. The errRouteLookupFailed was until now virtually a "const". Now that it depends on init data, we need a field on the Proxy struct.